packages/apps/human-app/frontend/src/modules/worker/components/profile/profile.page.tsx

```diff
@@ -28,9 +28,9 @@ export function WorkerProfilePage() {
   };
 
   useEffect(() => {
-    setNotifications();
+    if (!initializing) setNotifications();
     // eslint-disable-next-line react-hooks/exhaustive-deps -- call this once
-  }, [isConnected]);
+  }, [isConnected, initializing]);
```

Comment:
Could you please refactor this effect to be something like:

```
if (initializing) return;

if (!isConnected || !user.wallet_address) {
  showNotification...
}
```

Right now it's a bit tricky to understand in what cases notification is shown